rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -154,941 +218,203 @@ impl Handler<GetPhyIpFromMeshIp> for TunnelManager {
     }
 }
 
-pub struct UnListen(pub String);
-impl Message for UnListen {
-    type Result = ();
-}
-
-impl Handler<UnListen> for TunnelManager {
-    type Result = ();
-
-    fn handle(&mut self, un_listen: UnListen, _: &mut Context<Self>) -> Self::Result {
-        self.listen_interfaces.remove(&un_listen.0);
-        SETTING.get_network_mut().peer_interfaces = self.listen_interfaces.clone();
-    }
-}
-
-pub struct GetListen;
-impl Message for GetListen {
-    type Result = Result<HashSet<String>, Error>;
-}
-
-impl Handler<GetListen> for TunnelManager {
-    type Result = Result<HashSet<String>, Error>;
-    fn handle(&mut self, _: GetListen, _: &mut Context<Self>) -> Self::Result {
-        Ok(self.listen_interfaces.clone())
-    }
-}
-
 pub struct GetNeighbors;
 impl Message for GetNeighbors {
     type Result = Result<Vec<(LocalIdentity, String, IpAddr)>, Error>;
 }
 
 impl Handler<GetNeighbors> for TunnelManager {
-    type Result = ResponseFuture<Vec<(LocalIdentity, String, IpAddr)>, Error>;
+    type Result = Result<Vec<(LocalIdentity, String, IpAddr)>, Error>;
```

Comment:
Again, probably we want a set

Reply:
Also, isn't a cloned tunnels set/vec what we could just return? A `memcpy` would be quicker than the iteration below, and a lock on a reference would be even quicker that.

Reply:
this is a point of integration with traffic watcher which is IMO out of scope to modify with this pull.

Reply:
Conclusion from appear: this will be handled in a refactor issue @jkilpatr made